### PR TITLE
added clipShapeOnSave parameter

### PR DESCRIPTION
**Pull request by g0dzillaa**
Currently the app will clip the area outside of the Circle when we use the Circle shape. This is a problem if we only want to show the Circle shape but save the Rectangle shape in the server.

For example, if we want to show the avatar in circle shape and in the future if they want to show it in rectangle or need a rectangle image somewhere else in the platform, they cannot get it now as we are clipping the area outside of the circle.

`clipShapeOnSave` param allows us to save the image file in Rectangle path so that we can have the flexibility of changing the shape in the future. By default, the `clipShapeOnSave` is set to `true` to avoid any breaking changes.

### CODE REVIEW

File: lib/src/widgets/custom_image_crop_widget.dart [hunk 1]
```diff
@@ -250,16 +255,28 @@ class _CustomImageCropState extends State<CustomImageCrop>
     addTransition(CropImageData(x: event.delta.dx, y: event.delta.dy));
   }
 
-  Path _getPath(double cropWidth, double width, double height) {
+  Path _getPath(double cropWidth, double width, double height, bool clip) {
     switch (widget.shape) {
       case CustomCropShape.Circle:
+        if (clip) {
         return Path()
```
Comment (ikbendewilliam) [inline lib/src/widgets/custom_image_crop_widget.dart:262]: formatting, please use `dart format .`

Comment (ikbendewilliam) [reply lib/src/widgets/custom_image_crop_widget.dart:262]: I would also do the check to clip before the `switch`. This way if we add more shapes, you don't need to add the clip everywhere

**Review by ikbendewilliam — CHANGES_REQUESTED** [review]
Thank you for your PR! Just one small issue and formatting.

**Comment by g0dzillaa** [comment]
Closing this one as I mixed up some local files and too much hassle to fix it. Will open another PR.

**Pull request closed without merging**